### PR TITLE
IIIF Presentation

### PR DESCRIPTION
**Pull request by barmintor**

### CODE REVIEW

File: app/models/manifest.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    manifest_processing_file_path = path + '.processing'
+
+    # If progress file exists, return
+    return if File.exists?(manifest_processing_file_path)
+
+    # Touch file to block concurrent processes from trying to create an access copy
+    FileUtils.touch manifest_processing_file_path
+
+    begin
+      is_fs_mounted = false # documentary
```
Comment (elohanlon) [inline app/models/manifest.rb]: Not sure what this line documents.

File: Gemfile [hunk 2]
```diff
@@ -14,8 +14,8 @@ gem 'uglifier', '>= 1.3.0'
 # Use CoffeeScript for .coffee assets and views
 gem 'coffee-rails', '~> 4.1.0'
 # See https://github.com/rails/execjs#readme for more supported runtimes
-gem 'therubyracer', '>= 0.12.2',  platforms: :ruby
-gem 'libv8', '>= 3.16.14.15' # Min version for Mac OS 10.11, XCode 8.0
+gem 'mini_racer',  platforms: :ruby
```
Comment (elohanlon) [inline Gemfile:17]: I forgot, did you say that mini_racer didn't actually solve the problem here that you were trying to solve?

If that's the case, can I recommend this? (which is what we use in a couple of other apps)
```
gem 'therubyracer', '>= 0.12.3',  platforms: :ruby
gem 'libv8', '>= 3.16.14.19' # Min version for Mac OS 10.11, XCode 9.0, Ruby 2.4
```

Comment (barmintor) [reply Gemfile:17]: oh, it actually did make a difference - it installs!

File: app/models/manifest.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    format: 'jpg'
+  }
+  attr_reader :route_helper
+  def initialize(id_or_fedora_obj, route_helper)
+    super(id_or_fedora_obj)
+    @route_helper = route_helper
+  end
+
+  def queue_manifest_generation(queue_name = Derivativo::Queue::HIGH)
+    base_url = route_helper.iiif_id_url(id: 'do_not_use', version: THUMBNAIL_OPTS[:version])
```
Comment (elohanlon) [inline app/models/manifest.rb:22]: If I'm understanding the code correctly, only the host, protocol, and port portions of this base_url are important, right?  In that case, can we chop off the unhelpful portion of the URL before we pass it to the CreateManifestJob on the line beneath this one?  That way we won't have lots of "do_not_use" strings in the arguments of our job queue, which could seem confusing/incorrect later on (even though they wouldn't have any real effect).

Comment (barmintor) [reply app/models/manifest.rb:22]: I think that should do it.

Comment (elohanlon) [reply app/models/manifest.rb:22]: Thank you for all of your work!  Approving the PR now!

**Review by elohanlon — APPROVED** [review]

**Pull request merged**